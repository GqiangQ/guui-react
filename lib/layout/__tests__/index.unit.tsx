import {configure} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {Main, Layout, Header, Footer, Aside} from '../index';

configure({adapter: new Adapter()});

describe('Layout', () => {
  it('combines Main/Header/Footer/Aside', () => {
    const tree = renderer
      .create(
        <Layout>
          <Header></Header>
          <Layout dir="horizontal">
            <Aside></Aside>
            <Main></Main>
          </Layout>
          <Footer></Footer>
        </Layout>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('combines Main/Header/Footer/Aside in another way', () => {
    const tree = renderer
      .create(
        <Layout>
          <Aside></Aside>
          <Layout>
            <Header></Header>
            <Main></Main>
            <Footer></Footer>
          </Layout>
        </Layout>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
