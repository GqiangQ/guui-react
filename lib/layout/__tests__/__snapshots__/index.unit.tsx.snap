// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout combines Main/Header/Footer/Aside 1`] = `
<div
  className="gu-layout gu-layout-vertical "
>
  <div
    className="gu-header "
  />
  <div
    className="gu-layout gu-layout-horizontal "
  >
    <div
      className="gu-aside "
    />
    <div
      className="gu-main "
    />
  </div>
  <div
    className="gu-footer "
  />
</div>
`;

exports[`Layout combines Main/Header/Footer/Aside in another way 1`] = `
<div
  className="gu-layout gu-layout-vertical "
>
  <div
    className="gu-aside "
  />
  <div
    className="gu-layout gu-layout-vertical "
  >
    <div
      className="gu-header "
    />
    <div
      className="gu-main "
    />
    <div
      className="gu-footer "
    />
  </div>
</div>
`;
